File: python/bob/bob.py
#
# Skeleton file for the Python "Bob" exercise.
#

import re

def shoutp(what):
    if re.search('[A-Z]', what) and what == what.upper():
        return True
    else:
        return False

def questionp(what):
    if re.match('.+\?\s*$', what):
        return True
    else:
        return False

def silentp(what):
    if re.sub('\s', '', what) == '':
        return True
    else:
        return False


def hey(what):
    if shoutp(what):
        resp = 'Whoa, chill out!'
    elif questionp(what):
        resp = 'Sure.'
    elif silentp(what):
        resp = 'Fine. Be that way!'
    else:
        resp = 'Whatever.'
    return resp

File: javascript/bob/bob.js
//
// This is only a SKELETON file for the "Bob" exercise. It's been provided as a
// convenience to get you started writing code faster.
//

var Bob = function() {};

Bob.prototype.hey = function(input) {
    var shoutp = function(str) {
            /[A-Z]/.test(str) && str == str.toUpperCase;
    };
    var questionp = function(str) {
            /^.+\?$/.test(str);
    };
    var silentp = function(str) {
        str == str.trim();
    };

    if(shoutp(input)) {
        return 'Whoa, chill out!';
    } else if(questionp(input)) {
        return 'Sure';
    } else if(silentp(input)) {
        return 'Fine. Be that way!';
    } else {
        return 'Whatever.';
    };
};

module.exports = Bob;
